packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,670 @@
+   */
+  readonly monitoring?: Monitoring;
+
+  /**
+   * The tags assigned to job runs.
+   *
+   * @default - None
+   */
+  readonly tags?: { [key: string]: string };
+}
+
+/**
+ * Starts a job run.
+ *
+ * A job is a unit of work that you submit to Amazon EMR on EKS for execution.
+ * The work performed by the job can be defined by a Spark jar, PySpark script, or SparkSQL query.
+ * A job run is an execution of the job on the virtual cluster.
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-emr-eks.html
+ */
+export class EmrContainersStartJobRun extends sfn.TaskStateBase implements iam.IGrantable {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.RUN_JOB,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  public readonly grantPrincipal: iam.IPrincipal;
+  private role: iam.IRole;
+  private readonly logGroup?: logs.ILogGroup;
+  private readonly logBucket?: s3.IBucket;
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: EmrContainersStartJobRunProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.RUN_JOB;
+    validatePatternSupported(this.integrationPattern, EmrContainersStartJobRun.SUPPORTED_INTEGRATION_PATTERNS);
+
+    if (props.applicationConfig) {
+      this.validateAppConfigLength(this.props.applicationConfig);
+    }
+
+    if (props.jobDriver.sparkSubmitJobDriver?.entryPoint
+      && !sfn.JsonPath.isEncodedJsonPath(props.jobDriver.sparkSubmitJobDriver?.entryPoint.value)
+      && (props.jobDriver.sparkSubmitJobDriver?.entryPoint.value.length > 256
+        || props.jobDriver.sparkSubmitJobDriver?.entryPoint.value.length < 1)) {
+      throw new Error(`Entry point must be between 1 and 256 characters in length. Received ${props.jobDriver.sparkSubmitJobDriver?.entryPoint.value.length}.`);
+    }
+
+    if (props.jobDriver.sparkSubmitJobDriver?.entryPointArguments) {
+      if (!this.isArrayOfStrings(props.jobDriver.sparkSubmitJobDriver.entryPointArguments.value)
+      || (typeof props.jobDriver.sparkSubmitJobDriver.entryPointArguments.type === 'string'
+      && !sfn.JsonPath.isEncodedJsonPath(props.jobDriver.sparkSubmitJobDriver.entryPointArguments.value))) {
+        throw new Error(`Entry point arguments must be a string array. Received ${props.jobDriver.sparkSubmitJobDriver.entryPointArguments.type}.`);
+      }
+    }
+
+    if (props.jobDriver.sparkSubmitJobDriver?.entryPointArguments
+        && (props.jobDriver.sparkSubmitJobDriver.entryPointArguments.value.length > 10280
+          || props.jobDriver.sparkSubmitJobDriver.entryPointArguments.value.length < 1)) {
```

Comment:
this should include a check for `isArrayOfStrings` or we're not guaranteed that `.length` exists as a property or actually means "array length" (as opposed to string length, for example)